vault/src/main/java/org/opencadc/vault/NodePersistenceImpl.java

```diff
@@ -265,6 +297,37 @@ public ContainerNode getRootNode() {
         return root;
     }
 
+    @Override
+    public boolean isAllocation(ContainerNode cn) {
+        if (cn.parent == null) {
+            return false; // root is never an allocation
+        }
+        ContainerNode p = cn.parent;
+        for (ContainerNode ap : allocationParents) {
+            if (absoluteEquals(p.parent, ap)) {
```

Comment:
Isn't it sufficient to check the paths in order to conclude that it's the same node, e.g.  `(Utils.getPath(ap).equals(Utils.getPath(p.parent)))`?

Reply:
maybe. I generally don't like relying on string comparisons.

it turns out that in `vault` I could just use the ContainerNode uuid and not look at the path at all. I wrote this code in cavern and copied it, so it works but is more complex than required.

will change to use the uuids and see how it looks.